src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWflow.py

```diff
@@ -73,7 +209,12 @@ def docSchema(self):
             ('IsLogDBClean', False, bool),
             ('IsArchivalDelayExpired', False, bool),
             ('ForceArchive', False, bool),
-            ('RequestTransition', [], list)]
+            ('RequestTransition', [], list),
+            ('IncludeParents', False, bool),
+            ('DataPileup', None, (str, unicode)),
```

Comment:
And I just noticed these default values don't match the default values in the unit test `testTaskChainDefaults`.
DataPileup, MCPileup and ParentDataset are inconsistent.

Reply:
Yes, I have not yet started paying attention to the final version of the unit tests. I was about to do that in a later stage. Once we agree on the implementation of all the rest. Because we kind of depend on that for the unit tests step.

Reply:
Done

Reply:
Isn't the DataPileup/MCPileup/ParentDataset default value supposed to be a list instead of None?

Reply:
I think everything is as it should be, you just forget about this:

https://github.com/dmwm/WMCore/pull/10224/files/2f4a6275b7e0af14e3f42aabbca6f40470503ca9#diff-414c04814f1133dc334c7718010c11e70809389d23ec30e9d00d5e8874b84bf5R149-R152

Reply:
That's precisely my point. This data structure defines the schema of the workflow object used throughout MSRuleCleaner. By just looking at this, I'd expect those keys to have a value equal to `None`, not an empty list.

Does the whole logic - besides that explicit code you just referred to - change if you set default value to `[]` instead of None? If not and we get that for free, then I'd suggest to update the default value to make it consistent with the outcome of this workflow parser.

Reply:
It should work, but what would happen is that this conversion from a string to list will happen silently in the workflow parser class here: [1] instead of explicitly in the three lines I previously mentioned. I have tested it both ways, and now I do not remember which exactly but there was a corner case in which the silent conversion of the data structure was not preferable. So I preferred in the `-wfParser()` method to make the best effort to revert the structure back to the original one as described in the document template (which reflects the values as we expect them from reqmgr), and only then make the explicit conversion to lists only for those I really want and make them obvious in the `init()` method of MSRuleCleanerWflow.

[1]
https://github.com/dmwm/WMCore/pull/10224/files/2f4a6275b7e0af14e3f42aabbca6f40470503ca9#diff-414c04814f1133dc334c7718010c11e70809389d23ec30e9d00d5e8874b84bf5R110-R111

Reply:
Actually, I just tested it again.... And not even a corner case but a rather general one. It tries to create a list of all letters present in all the MCPilup dataset names it can find and explodes.

Reply:
Okay, let it be then. Thanks for checking this out Todor.